### PR TITLE
Rework the game

### PR DESCRIPTION
**Pull request by athei**
As discussed. This will require the players to submit their own turns. Every contract can only submit one turn per block. I tried to add all the getters that are necessary for the UI.

### CODE REVIEW

File: game/lib.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 #[ink::contract]
-mod game {
+mod sqink_splash {
```
Comment (DoubleOTheven) [inline game/lib.rs:4]: haha. love it

File: game/lib.rs [hunk 2]
```diff
@@ -10,229 +10,282 @@ mod game {
...
-    const GAS_ALLOWANCE_PER_ROUND: i32 = 750_000_000;
+    /// How much score should be addded per field that is occupied by a user.
+    const SCORE_PER_FIELD: u64 = 1_000_000_000;
 
     #[ink(storage)]
-    pub struct Game {
-        /// Privileged account to control the game.
+    pub struct SquinkSplash {
+        /// Owner of the contract. Can initiate a new game.
         admin: AccountId,
```
Comment (DoubleOTheven) [inline game/lib.rs:35]: nice, i was going to do this

File: game/lib.rs (hunk 1 above)
Comment (cmichi) [inline game/lib.rs:4]:
```suggestion
mod squink_splash {
```

File: game/lib.rs [hunk 3]
```diff
@@ -10,229 +10,282 @@ mod game {
...
-    pub struct Scoreboard {
-        scoreboard: Vec<(PlayerId, Score)>,
-    }
-
-    #[ink(event)]
-    #[derive(Debug)]
-    pub struct PaintedPixel {
-        player: AccountId,
-        pixel: (X, Y),
+    pub struct Player {
```
Comment (DoubleOTheven) [inline game/lib.rs:64]: I think we should user names for the player contract too. It will be easier to identify the players (and yourself). We can do it in another PR. Just mentioning it

**Review by DoubleOTheven — APPROVED** [review]

**Review by cmichi — APPROVED** [review]
Awesome, thanks!

File: game/lib.rs (hunk 3 above)
Comment (athei) [reply game/lib.rs:68]: Good idea. Added.
